sql-cli/sql_cli/dag_generator.py

```diff
@@ -83,6 +88,45 @@ def sorted_sql_files(self) -> list[SqlFile]:
 
         return list(depth_first_search.dfs_postorder_nodes(graph))
 
+    def to_transform_task_list(self) -> list[TransformOperator]:
+        """
+        Converts the list of SQL Files into a list of TranformOperator tasks
+        that all have proper dependencies set.
+        :return:
+        """
+        param_dict = {s.path.stem: s.to_transform_operator() for s in self.sql_files}
+        for s in self.sql_files:
+            for p in s.get_parameters():
+                param_dict[s.path.stem].parameters[p] = param_dict[p].output
+                param_dict[s.path.stem].set_upstream(param_dict[p])
+        return list(param_dict.values())
+
+
+def render_dag(directory: Path, workflow_name: str, start_date: datetime = datetime(2020, 1, 1)) -> DAG:
+    """
+    render_dag allows a user to take any directory and turn it into a runnable
+    Airflow DAG. This function will read all SQL files, and set dependencies based
+    on jinja template-based variables.
+    :param directory: Base directory for SQL files. We will recursively parse
+        subdirectories as well.
+    :param workflow_name: the name of the Workflow you would like to run
+    :param start_date: (Optional) the start date you would like to set for your run.
+        defaults to 2020-01-01
+    :return: a DAG that can be run by any airflow
+    """
+    if not directory.exists():
+        raise SqlFilesDirectoryNotFound("The directory does not exist!")
+    sql_files: list[SqlFile] = sorted(get_sql_files(directory, target_directory=None))
+    sql_files_dag = SqlFilesDAG(
+        dag_id=directory.name,
+        start_date=datetime(2020, 1, 1),
+        sql_files=sql_files,
+    )
+    dag = DAG(dag_id=workflow_name, start_date=start_date)
```

Comment:
Let's use the information from the SqlFilesDAG. WDYT?
```suggestion
    dag = DAG(dag_id=sql_files_dag.dag_id, start_date=sql_files_dag.start_date)
```